packages/jest-cli/src/reporters/coverage_reporter.js

```diff
@@ -224,10 +225,59 @@ class CoverageReporter extends BaseReporter {
           return errors;
         }, []);
       }
-      const errors = check(
-        'global',
-        globalConfig.coverageThreshold.global,
-        results,
+
+      const thresholdConfigFilePaths = Object.keys(
+        globalConfig.coverageThreshold,
+      )
+        .filter(thresholdKey => thresholdKey !== 'global')
+        .map(thresholdKey => path.resolve(thresholdKey));
+      const filteredCoverageSummary = map
+        .files()
+        .filter(filePath => thresholdConfigFilePaths.indexOf(filePath) === -1)
+        .map(filePath => map.fileCoverageFor(filePath))
+        .reduce((summary: ?CoverageSummary, fileCov: FileCoverage) => {
+          if (summary !== undefined && summary !== null) {
+            summary.merge(fileCov.toSummary());
+          } else {
+            summary = fileCov.toSummary();
+          }
+          return summary;
+        }, undefined);
+
+      const errors = [].concat(
+        ...Object.keys(globalConfig.coverageThreshold)
```

Comment:
Mixing array concat and spread. For now just use concat, because we're not transforming array spreading (not supported on Node 4, that's why Circle fails)